rendering.js

```diff
@@ -6,16 +6,49 @@ const toPx = (value) => `${value}px`
 const toPercent = (value) => `${value}%`
 
 // __This is the main render function which delegates control to the more-specific render functions.__
-export function renderGame(state) {
+export function renderGame(prevState, state) {
+    function hasStateChanged(stateSelectors) {
+        for (const selector of stateSelectors) {
+            if (selector(prevState) !== selector(state)) {
+                return true
+            }
+        }
+        return false
+    }
+
     renderFallingObjects(state)
-    renderLivesRemaining(state)
-    renderBasketValue(state)
-    renderOverlay(state)
-    renderButtons(state)
-    renderBasket(state)
-    renderScore(state)
-    renderLevel(state)
-    renderSounds(state)
+
+    if (hasStateChanged([(s) => s.livesRemaining])) {
```

Comment:
Thanks for the feedback Tim!

`s` is state. I agree, it's lacking of description, though my argument when I wrote it was that:
1. It might be obvious from context: the function it's passed into is called `hasStateChanged`
2. the name `state` is already used in the surrounding scope. It's actually valid JS to re-use the names from outer scopes for different vars in inner scopes, but it's an easy way to end up with bugs if you start reorganizing code. 

I might rename that `state` in the outer context to `nextState` and then use `state` in the inner one, then everything's more clear

Reply:
Well because this s state is only used here, maybe just put a comment above that tells us s is for state in this local scope. I was guessing that it was state, but couldn't be 100% sure. If you call it next state, remember that there is nextState used elsewhere, which could possibly bring up the same bugs you mentioned in your point 2

Reply:
I just added the comment